### ISSUE TITLE
vlc rtsp on nixos does not work

### ISSUE THREAD

**Issue by wwat** [NixOS/nixpkgs#5573]
On the same machine I can use /usr/bin/vlc (Ubuntu 14.04) to access the following URL (and see a camera feed), but using ~/.nix-profile/bin/vlc I get the bottom message. 
```
rtsp://root:@192.168.1.152/h264
```
main input error: open of `rtsp://root:@192.168.1.152/h264' failed

```
Ubuntu version which works: VLC media player 2.1.4 Rincewind (revision 2.1.4-0-g2a072be)
```

```
NixOS version which is broken: VLC media player 2.1.5 Rincewind (revision 2.1.4-49-gdab6cb5)
The version on my native NixOS is the same: VLC media player 2.1.5 Rincewind (revision 2.1.4-49-gdab6cb5)
```

**Comment by bjornfor** [comment]
Can you run strace on that? I'm guessing there is a missing library that needs to be given to vlc at build time.

**Comment by wwat** [comment]
@bjornfor

strace was mostly useless, but the -l option proved useful:
```
diff a b                                                                                                                                 
2a3,5
>   access_realrtsp        Real RTSP
>   live555                RTP/RTSP/SDP demuxer (using Live555)
>   live555                RTSP/RTP access and demux
```
TLDR; the live555 library is missing and I don't know what access_realrtsp does.

**Comment by bjornfor** [comment]
live555 isn't in nixpkgs (yet). Maybe you could package it?

**Comment by wwat** [comment]
@bjornfor I packaged live555 (at least, something ended up in the nix store) (with that name in some other nixpkgs root R), and I modified the expression for vlc (default.nix) to also depend on `live555`, but how do I actually make it such that this ties in with the other expression which depends on `vlc`. I want that `vlc` to really be the modified `vlc`. 

I'd prefer you to be very specific.

**Comment by wwat** [comment]
In particular, I don't get why the following doesn't work in `configuration.nix`: 
```
  packageOverrides = pkgs: rec {
         # instead of the standard vlc, use mine
         vlc = unstable.vlc;
         };
 ```

**Comment by bjornfor** [comment]
Maybe ```unstable``` isn't what you think it is? Other than that it looks correct.

Try this:

```
    packageOverrides = pkgs: {
      vlc = (import /path/to/modified-nixpkgs/default.nix { }).vlc;
    };
```

**Comment by wwat** [comment]
@bjornfor It says that packageOverrides is not an attribute of the body in configuration.nix. Why? I have no idea. I did manage to start a build of vlc, which probably means that unstable.vlc (which has my modifications) is running.

**Comment by wwat** [comment]
Building vlc goes at glacial speed, btw. Why doesn't it use all of my cores?

**Comment by bjornfor** [comment]
Ah, you need to put the packageOverrides thing in the nixpkgs.config option. Paste this snippet into configuration.nix:

```
  nixpkgs.config = {
    packageOverrides = pkgs: {
      vlc = (import /path/to/modified-nixpkgs/default.nix { }).vlc;
    };
  };
```

**Comment by bjornfor** [comment]
Ref. http://nixos.org/nixos/manual/ch-options.html#opt-nixpkgs.config

**Comment by bjornfor** [comment]
Per-package building is single-threaded by default, because so many upstream packages fail randomly with parallel builds. Once you build more packages in parallell, nix will use all your cores.

If you want to build vlc itself with all your cores, set ```enableParallelBuilding = true;``` in the vlc expression.

There is a ML thread or github issue (don't remember) about making that flag a global option too, so that adventurous users can get max speed (at the expense of possibly random failures).

**Comment by wwat** [comment]
The expression already contains `enableParallelBuilding = true` (the 14.12 one already did).

**Comment by bjornfor** [comment]
Then I don't know. Try adding "set -x" in a preConfigure phase or something and look at what commands are being run.

**Comment by peti** [comment]
`/etc/nix/nix.conf` must contain a setting like 

    build-cores = 0  # auto-detect number of cores

for parallel building to take any effect.

**Comment by wwat** [comment]
This information is already contained in `hardware-configuration.nix` (via `nix.maxJobs`). Why does `build-cores` even exist?

**Comment by vcunat** [comment]
You can also override this per-build by the ``--cores`` command-line parameter.

**Comment by bjornfor** [comment]
Ah, now I learned something new! For one, I thought the default was auto-detection of build-cores, but it is 1 by default, according to "man configuration.nix". The documentation doesn't mention the special '0' value though.

**Comment by vcunat** [comment]
The docs for ``--cores`` and ``build-cores`` do mention what 0 means (``buildCores`` doesn't). Perhaps such parts of docs should be linked somehow, so we don't have to duplicate the information.

**Comment by peti** [comment]
I've extended the docs in ea94a6d.

**Comment by peti** [comment]
By the way, can we close this ticket? Or is there still an open question?

**Comment by wwat** [comment]
I think we should close it when live555 support is in. (I think I have a working version, but I don't have access to it for some time. )

**Comment by wwat** [comment]
Thank you for the cooperation.
